src/meta/attribute.rs

```diff
@@ -1500,14 +1500,31 @@ impl Preview {
         let width = u32::read(read)? as usize;
         let height = u32::read(read)? as usize;
 
-        let pixel_data = i8::read_vec(read, width * height * components_per_pixel, 1024*1024*4, None, "preview attribute pixel count")?;
+        if let Some(size) = width.checked_mul(height) {
+            if let Some(total_size) = size.checked_mul(components_per_pixel) {
```

Comment:
this variable (total_size) never had a nice name in the first place, could you rename it to `byte_count` while you're at it? Would be nice

Reply:
`byte_count` sounds good. Maybe change `size`  to `pixel_count`?

Reply:
Excellent, indeed